core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -224,7 +227,9 @@ private class ShuffleStatus(
     logDebug(s"Removing existing map output ${mapIndex} ${bmAddress}")
     if (mapStatuses(mapIndex) != null && mapStatuses(mapIndex).location == bmAddress) {
       _numAvailableMapOutputs -= 1
-      mapStatusesDeleted(mapIndex) = mapStatuses(mapIndex)
+      val currentMapStatus = mapStatuses(mapIndex)
```

Comment:
Pull `currentMapStatus` outside the `if`, and use it for the `if` condition as well.